backend/routes/orders.js

```diff
@@ -1,16 +1,61 @@
 const { authSellerAndShop } = require('./_auth')
-const { Order } = require('../models')
+const {
+  Order,
+  sequelize,
+  Sequelize: { Op }
+} = require('../models')
 const { findOrder } = require('../utils/orders')
 const makeOffer = require('./_makeOffer')
 const { sendNewOrderEmail } = require('../utils/emailer')
 
 module.exports = function (app) {
   app.get('/orders', authSellerAndShop, async (req, res) => {
-    const orders = await Order.findAll({
-      where: { shopId: req.shop.id },
-      order: [['createdAt', 'desc']]
+    const { page: pageVal, search } = req.query
+
+    const page = parseInt(pageVal) || 1
+    const limit = parseInt(process.env.ORDER_PAGINATION_LIMIT) || 50
+    const offset = page ? (page - 1) * limit : undefined
+
+    const where = {
+      shopId: req.shop.id
+    }
+
+    if (search) {
+      where[Op.or] = [
+        sequelize.where(
+          sequelize.cast(sequelize.col('data'), 'text'),
+          'ILIKE',
+          `%${search}%`
+        )
+      ]
+    }
+
+    const { count, rows: orders } = await Order.findAndCountAll({
+      where,
+      order: [
+        // ['createdAt', 'desc'],
+        [
+          sequelize.cast(
+            sequelize.fn('split_part', sequelize.col('order_id'), '-', 4),
```

Comment:
nit: can you add a comment explaining how this ordering works and why we are not simply ordering by createdAt? Also note that the performance will get bad for shops that have large number of orders since no index can be used when calling a SQL function like split_part. Not a concern right now given our order volume is tiny but worth mentioning for later as a TODO.